test/python/visualization/test_plot_histogram.py

```diff
@@ -107,6 +107,18 @@ def test_different_counts_lengths(self):
         fig = plot_histogram([raw_dist, exact_dist])
         self.assertIsInstance(fig, mpl.figure.Figure)
 
+    def test_with_number_to_keep(self):
+        """Test plotting using number_to_keep"""
+        dist = {"00": 3, "01": 5, "11": 8, "10": 11}
+        fig = plot_histogram(dist, number_to_keep=2)
+        self.assertIsInstance(fig, mpl.figure.Figure)
+
+    def test_with_number_to_keep_multiple_executions(self):
```

Comment:
This test doesn't actually check that the issue is resolved, it just runs the improved code with no checks. Personally, I'd prefer an additional test to run ``_plot_histogram_data()`` and validate its results.

Reply:
@drewrisinger Thank you for your review! 
In the case of this issue, plot_histogram was throwing an exception. After I fixed the specified case, I noticed that the function may receive multiple distributions and that case was again throwing an exception. 
_plot_histogram_data() is not part of the public API and I don't have access to it.

Reply:
I understand that technically it should be possible, but somehow it doesn't work. I see that your approach with comparing the files is used in multiple other tests and maybe it should be used here as well.